router/router.go

```diff
@@ -59,7 +59,7 @@ func (router *Router) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// isHTTPEvent checks if a request carries HTTP event. It also accepts pre-flight CORS requests because CORS is
 	// resolved downstream.
 	if isHTTPEvent(r) {
-		metricEventsHTTPReceived.Inc()
+		metricEventsReceived.WithLabelValues("", "http").Inc()
 
 		event, _, err := router.eventFromRequest(r)
 		if err != nil {
```

Comment:
Would it be easier to put `metricEventsProcessed.WithLabelValues(space, "http").Inc()` after line 72 below, rather than in the `handleHTTPEvent` method? Follows the pattern of how the `invoke` event is done, plus it keeps it together a little better.

Wouldn't let me comment on line 72 directly since it wasn't in the diff.

Reply:
Yeah, that would make sense but I don't want to return space from `handleHTTPEvent`. There is some refactoring needed here. Metrics reporting could also be moved to inside those handle* functions but again there will be exception for HTTP event. Let's leave it as it is right now as there is no ideal solution.